front/src/applications/stdcm/utils/formatSimulationReportSheet.ts

```diff
@@ -106,6 +130,121 @@ export function getStopDurationBetweenTwoPositions(
   return null;
 }
 
+/**
+ * @param op Operational point to format
+ * @param train Object containing simulated train positions, times, and departure time
+ * @param simulationPathSteps List of simulation path steps
+ * @returns A formatted operational point with calculated stop duration and departure time
+ */
+function formatOperationalPointWithTimes(
+  op: SuggestedOP,
+  train: TrainSimulationProps,
+  simulationPathSteps: StdcmPathStep[]
+): StdcmResultsOperationalPoint {
+  const formattedTime = getTimeAtPosition(op.positionOnPath, train);
+
+  const duration = getStopDurationAtPosition(op.positionOnPath, train.positions, train.times);
+  const durationInSeconds = duration !== null ? duration / 1000 : 0;
+  const durationToString = secondsToTimeString(durationInSeconds);
+  const stopEndTime = computeStopDepartureTime(formattedTime, durationToString);
+  // Find the corresponding stopType from pathSteps
+  const correspondingStep = simulationPathSteps.find(
+    (step) => step.location && matchPathStepAndOp(step.location, op)
+  );
+  let stopType;
+  if (correspondingStep) {
+    stopType = !correspondingStep.isVia ? 'serviceStop' : correspondingStep.stopType;
```

Comment:
The ternary could be more readable if we reverse the condition. Also, let's take advantage of TypeScript's type safety by using the `StdcmStopTypes` enum instead of string literals:
```suggestion
    stopType = correspondingStep.isVia ? correspondingStep.stopType : StdcmStopTypes.SERVICE_STOP;
```

Reply:
Sure ^^